src/IECoreHoudini/SOP_SceneCacheSource.cpp

```diff
@@ -305,6 +308,70 @@ OP_ERROR SOP_SceneCacheSource::cookMySop( OP_Context &context )
 	return error();
 }
 
+ConstObjectPtr SOP_SceneCacheSource::buildTransformPointCloud( const IECoreScene::SceneInterface *scene,const Imath::M44d transform,const double time, Parameters &params,const bool hasObject )
+{
+	params.hasAnimatedTopology = false;
+	params.hasAnimatedPrimVars = true;
+	params.animatedPrimVars.clear();
+	Imath::V3f p(0);
+	if ( hasObject )
+	{
+		p = scene->readBound( time ).center();
+	}
+	std::vector<Imath::V3f> point( 1, p );
+	PointsPrimitivePtr points = new PointsPrimitive( new V3fVectorData( point ) );
+
+	// has object
+	std::vector<int> hasObjects( 1, hasObject );
+	points->variables["hasshape"] = PrimitiveVariable( PrimitiveVariable::Vertex, new IntVectorData( hasObjects ) );
```

Comment:
Can we call this primvar `hasObject` to match the API? There is no reason the object must be a "shape" (primitive).